collect_app/src/test/java/org/odk/collect/android/database/DatabaseMediaFileRepositoryTest.java

```diff
@@ -15,6 +18,32 @@ public void whenThereIsNoFormForgivenFormIdAndVersion_shouldGetAllMethodReturnEm
         FormsDao formsDao = mock(FormsDao.class);
         when(formsDao.getFormMediaPath("1", "1")).thenReturn(null);
 
-        assertThat(new DatabaseMediaFileRepository(formsDao).getAll("1", "1").size(), is(0));
+        assertThat(new DatabaseMediaFileRepository(formsDao, new FileUtil()).getAll("1", "1").size(), is(0));
+    }
+
+    @Test
+    public void whenMediaFolderFileDoesNotExist_shouldGetAllMethodReturnEmptyArray() {
+        FormsDao formsDao = mock(FormsDao.class);
+        FileUtil fileUtil = mock(FileUtil.class);
+        File file = mock(File.class);
+        String formMediaPath = "/samplePath/";
+
+        when(formsDao.getFormMediaPath("1", "1")).thenReturn(formMediaPath);
+        when(fileUtil.getFileAtPath(formMediaPath)).thenReturn(file);
+        when(file.exists()).thenReturn(false);
+
+        assertThat(new DatabaseMediaFileRepository(formsDao, new FileUtil()).getAll("1", "1").size(), is(0));
+    }
+
+    @Test
+    public void whenMediaFolderFileIsNull_shouldGetAllMethodReturnEmptyArray() {
+        FormsDao formsDao = mock(FormsDao.class);
+        FileUtil fileUtil = mock(FileUtil.class);
+        String formMediaPath = "/samplePath/";
+
+        when(formsDao.getFormMediaPath("1", "1")).thenReturn(formMediaPath);
+        when(fileUtil.getFileAtPath(formMediaPath)).thenReturn(null);
```

Comment:
I could be missing something but I don't understand how this case can happen? `FileUtil#getFileAtPath` looks like it couldn't return `null`.

Reply:
Theoretically you are right but after fixing bugs like https://github.com/getodk/collect/issues/4102 I'm trying to check more than usually. Is that bad? Those are just tests so I think it's ok to tests **more** sometimes.

Reply:
Yeah true. I guess I'm thinking about if we've not mocked `FileUtil` as we have we can't guarantee that the method won't return `null` from this test.